### DISCUSSION TITLE
Error When Calculating Polygenic Risk Scores

### DISCUSSION THREAD

**Discussion by siyagoel** [PGScatalog/pgsc_calc#143] [Q&A]
Hi! I am running this line of code:

nextflow run pgscatalog/pgsc_calc -profile docker  --input samplesheet.csv  --target_build GRCh37 --pgs_id PGS001229

When this happens, the following errors pop up:
......................
NOTE: Process `PGSCATALOG_PGSCALC:PGSCALC:DOWNLOAD_SCOREFILES ([pgs_id:PGS001229, pgp_id:, trait_efo:])` terminated with an error exit status (137) -- Execution is retried (1)
ERROR ~ Error executing process > 'PGSCATALOG_PGSCALC:PGSCALC:MAKE_COMPATIBLE:PLINK2_VCF (stephen_vcf_subset chromosome ALL)'

Caused by:
  Process `PGSCATALOG_PGSCALC:PGSCALC:MAKE_COMPATIBLE:PLINK2_VCF (stephen_vcf_subset chromosome ALL)` terminated with an error exit status (6)

Tip: view the complete command output by changing to the process work dir and entering the command `cat .command.out`

 -- Check '.nextflow.log' file for details
ERROR ~ ERROR: No scores calculated!

 -- Check '.nextflow.log' file for details
....................
I was wondering if anyone know how to solve this issue? Thanks.

**Comment by nebfield** [comment]
Unfortunately there's not enough information to understand what went wrong. 

Did the `PLINK2_VCF` process print out any additional information? Could you try deleting the cache and re-running the workflow to see if there was a temporary problem?

```
$ rm -r work
$ nextflow run pgscatalog/pgsc_calc -profile docker --input samplesheet.csv --target_build GRCh37 --pgs_id PGS001229
```

**Reply by siyagoel** [reply]
Hi! Here is the detailed error:

[6a/655a0a] NOTE: Process `PGSCATALOG_PGSCALC:PGSCALC:DOWNLOAD_SCOREFILES ([pgs_id:PGS001229, pgp_id:, trait_efo:])` terminated with an error exit status (137) -- Execution is retried (1)
ERROR ~ Error executing process > 'PGSCATALOG_PGSCALC:PGSCALC:MAKE_COMPATIBLE:PLINK2_VCF (stephen_vcf_subset chromosome ALL)'

Caused by:
  Process `PGSCATALOG_PGSCALC:PGSCALC:MAKE_COMPATIBLE:PLINK2_VCF (stephen_vcf_subset chromosome ALL)` terminated with an error exit status (6)

Command executed:

  plink2 \
      --threads 2 \
      --memory 16384 \
      --set-all-var-ids '@:#:$r:$a' \
      --max-alleles 2 \
      --new-id-max-allele-len 100 missing \
      --vcf DTX86784.filtered.snp.vcf.gz  \
      --make-pgen vzs\
      --out vcf_stephen_vcf_subset_ALL # 'vcf_' prefix is important
  
  cat <<-END_VERSIONS > versions.yml
  "PGSCATALOG_PGSCALC:PGSCALC:MAKE_COMPATIBLE:PLINK2_VCF":
      plink2: $(plink2 --version 2>&1 | sed 's/^PLINK v//; s/ 64.*$//' )
  END_VERSIONS

Command exit status:
  6

Command output:
  --vcf: 3752k variants scanned.
  --vcf: 3753k variants scanned.
  --vcf: 3754k variants scanned.
  --vcf: 3755k variants scanned.
  --vcf: 3756k variants scanned.
  --vcf: 3757k variants scanned.
  --vcf: 3758k variants scanned.
  --vcf: 3759k variants scanned.
  --vcf: 3760k variants scanned.
  --vcf: 3761k variants scanned.
  --vcf: 3762k variants scanned.
  --vcf: 3763k variants scanned.
  --vcf: 3764k variants scanned.
  --vcf: 3765k variants scanned.
  --vcf: 3766k variants scanned.
  --vcf: 3767k variants scanned.
  --vcf: 3768k variants scanned.
  --vcf: 3769k variants scanned.
  --vcf: 3770k variants scanned.
  --vcf: 3771k variants scanned.
  --vcf: 3772k variants scanned.
  --vcf: 3773k variants scanned.
  --vcf: 3774k variants scanned.
  --vcf: 3775k variants scanned.
  --vcf: 3776k variants scanned.
  --vcf: 3777k variants scanned.
  --vcf: 3778k variants scanned.
  --vcf: 3779k variants scanned.
  --vcf: 3780k variants scanned.
  --vcf: 3781k variants scanned.
  --vcf: 3782k variants scanned.
  --vcf: 3783k variants scanned.
  --vcf: 3784k variants scanned.
  --vcf: 3785k variants scanned.
  --vcf: 3786k variants scanned.
  --vcf: 3787k variants scanned.
  --vcf: 3788k variants scanned.
  --vcf: 3789k variants scanned.
  --vcf: 3790k variants scanned.
  --vcf: 3791k variants scanned.
  --vcf: 3792k variants scanned.
  --vcf: 3793k variants scanned.
  --vcf: 3794k variants scanned.
  --vcf: 3795k variants scanned.
  --vcf: 3796k variants scanned.
  --vcf: 3797k variants scanned.
  --vcf: 3798k variants scanned.
  --vcf: 3799k variants scanned.
  --vcf: 3800k variants scanned.
  End time: Fri Aug 18 20:14:30 2023

Command error:
  --vcf: 3757k variants scanned.
  --vcf: 3758k variants scanned.
  --vcf: 3759k variants scanned.
  --vcf: 3760k variants scanned.
  --vcf: 3761k variants scanned.
  --vcf: 3762k variants scanned.
  --vcf: 3763k variants scanned.
  --vcf: 3764k variants scanned.
  --vcf: 3765k variants scanned.
  --vcf: 3766k variants scanned.
  --vcf: 3767k variants scanned.
  --vcf: 3768k variants scanned.
  --vcf: 3769k variants scanned.
  --vcf: 3770k variants scanned.
  --vcf: 3771k variants scanned.
  --vcf: 3772k variants scanned.
  --vcf: 3773k variants scanned.
  --vcf: 3774k variants scanned.
  --vcf: 3775k variants scanned.
  --vcf: 3776k variants scanned.
  --vcf: 3777k variants scanned.
  --vcf: 3778k variants scanned.
  --vcf: 3779k variants scanned.
  --vcf: 3780k variants scanned.
  --vcf: 3781k variants scanned.
  --vcf: 3782k variants scanned.
  --vcf: 3783k variants scanned.
  --vcf: 3784k variants scanned.
  --vcf: 3785k variants scanned.
  --vcf: 3786k variants scanned.
  --vcf: 3787k variants scanned.
  --vcf: 3788k variants scanned.
  --vcf: 3789k variants scanned.
  --vcf: 3790k variants scanned.
  --vcf: 3791k variants scanned.
  --vcf: 3792k variants scanned.
  --vcf: 3793k variants scanned.
  --vcf: 3794k variants scanned.
  --vcf: 3795k variants scanned.
  --vcf: 3796k variants scanned.
  --vcf: 3797k variants scanned.
  --vcf: 3798k variants scanned.
  --vcf: 3799k variants scanned.
  --vcf: 3800k variants scanned.
  Error: Invalid chromosome code 'GL000207.1' on line 3801064 of --vcf file.
  (Use --allow-extra-chr to force it to be accepted.)
  End time: Fri Aug 18 20:14:30 2023
  cp: .command.out and .command.out are identical (not copied).
  cp: .command.err and .command.err are identical (not copied).
  cp: .command.trace and .command.trace are identical (not copied).

Work dir:
  /Users/siyagoel/work/d9/7e72cdffdedf193157b27d05abd26f

Tip: when you have fixed the problem you can continue the execution adding the option `-resume` to the run command line

 -- Check '.nextflow.log' file for details
ERROR ~ ERROR: No scores calculated!

 -- Check '.nextflow.log' file for details

**Reply by smlmbrt** [reply]
Hi @siyagoel, it seems like your input VCF has non-standard chromosomes. If you follow the instructions in https://github.com/PGScatalog/pgsc_calc/issues/132#issuecomment-1665620571 adding `--allow-extra-chr` should fix this. 

This is a common problem so we'll try to handle it better in future releases (#159).

**Reply by siyagoellnew** [reply]
Hi! Thanks so much. I was wondering if I could meet with one of the maintainers/people who are experienced with this tool. I am trying to integrate this into AWS and am running into some issues. Thanks a lot.

**Reply by smlmbrt** [reply]
Are the issues the same as #108 and #155?

**Reply by siyagoellnew** [reply]
It is the same as 132. I edited both lines 41 and 53. However, I am still getting this error:

ERROR ~ Error executing process > 'PGSCATALOG_PGSCALC:PGSCALC:DOWNLOAD_SCOREFILES ([pgs_id:PGS001229, pgp_id:, trait_efo:])'

Caused by:
  Process `PGSCATALOG_PGSCALC:PGSCALC:DOWNLOAD_SCOREFILES ([pgs_id:PGS001229, pgp_id:, trait_efo:])` terminated with an error exit status (137)

Was wondering if I could meet with someone to help fix this. Thanks.

**Reply by bensaini717** [reply]
I am facing the same issues and would also find a working session helpful. Please share more info about this possible meeting.

Reading through the links that @smlmbrt has included in the post, I believe the problem primarily comes from the lack of a standard file format and the minimum required values. For example, how to identify an imputed VCF from a generic one. I wonder if someone has a created pipeline that takes a FASTq file and spits out a pgs_calc-compatible VCF file.

**Reply by smlmbrt** [reply]
@bensaini717 an imputed VCF will often have a dosage genotype field (`DS`, https://pgsc-calc.readthedocs.io/en/latest/how-to/samplesheet.html#setting-genotype-field) and be generated from an imputation server (e.g. Michigan, TopMed, Sanger). This is more common for larger numbers of samples that have been genotyped using arrays and not sequencing.
